### PR TITLE
Feature/swipe page

### PR DESCRIPTION
**Pull request by Ektoplasme**
### Adds Swipe Page

Creation of the Swipe component:

- I used [react-spring](https://react-spring.dev/) + [use-gesture](https://use-gesture.netlify.app/) to make the cards animations and drag events
- Added a query to get all the cats profile to swipe: only those we didn't like or dislike
- "Pagination" system, triggers 5 new cards redistribution when the stack is empty
- Added 2 mutations for like and dislike
- Added a modal popup that is triggered when a like mutation returns isMatch=true
- Updated overall style, layout, mobile format, added gifs...

azdazdazd-ezgif com-video-to-gif-converter

(Video is a gif, so it's normal that it's low fps)

### CODE REVIEW

**Review by NicolasSokolowski — DISMISSED** [review]
👍 👍 👍

File: client/src/components/SwipeCard/SwipeCard.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      sx={{
+        height: 460,
+        width: 380,
+        borderRadius: "40px",
+        boxShadow:
+          "0 12.5px 50px -10px rgba(50, 50, 73, 0.4), 0 10px 10px -10px rgba(50, 50, 73, 0.3)",
+      }}
+    >
+      <CardCover>
+        <img src={profile_picture} loading="lazy" alt="" />
```
Comment (Pas-nou) [inline client/src/components/SwipeCard/SwipeCard.tsx]: ton `alt=""` est vide ici

File: client/src/main.tsx [hunk 2]
```diff
@@ -3,7 +3,8 @@ import { createRoot } from "react-dom/client";
 import { CssVarsProvider } from "@mui/joy/styles";
 import { ApolloProvider } from "@apollo/client";
 import { createBrowserRouter, RouterProvider } from "react-router-dom";
-import { Home } from "./views/Home/Home";
+// import { Home } from "./views/Home/Home";
```
Comment (Pas-nou) [inline client/src/main.tsx]: 👀

**Review by Pas-nou — DISMISSED** [review]
2 petites choses à modifier, mais là t'as abbattu du gros travail 😸

**Review by pjroggy — DISMISSED** [review]

**Review by Pas-nou — DISMISSED** [review]

**Review by NicolasSokolowski — DISMISSED** [review]

**Review by pjroggy — DISMISSED** [review]

File: client/src/components/SwipeCard/SwipeCard.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      />
+      <CardContent sx={{ justifyContent: "flex-end" }}>
+        <Typography level="title-lg" textColor="#fff">
+          {name}, {age}
+        </Typography>
+        <Typography textColor="#fff">{surname}</Typography>
+        <ul className="swipe-card-interests-list">
+          {interests.map((interest, index) => {
+            return (
+              <li key={index}>
```
Comment (jujuck) [inline client/src/components/SwipeCard/SwipeCard.tsx]: Attention, pas d'index en key. Cela contourne les règles d'optimisation de rendu de react

File: server/src/cats/cat.resolvers.ts [hunk 4]
```diff
@@ -66,4 +66,95 @@ export default class CatResolver {
...
+      where: { cat_id1: likedCat, cat_id2: connectedCat },
+    });
+
+    // Créez une nouvelle instance de Like
+    const like = Like.create({
+      cat_id1: connectedCat,
+      cat_id2: likedCat,
+      isMatch: reciprocalLike ? true : null, // Si like reciproque match true, si pas réciproque le laisser à null
+    });
+
```
Comment (jujuck) [inline server/src/cats/cat.resolvers.ts:132]: Il manque une étape de réciprocité 
Si reciprocalLike, alors reciprocallike.isMatch = true puis save

File: server/src/cats/cat.resolvers.ts [hunk 5]
```diff
@@ -66,4 +66,95 @@ export default class CatResolver {
...
+    // Check si le chat liké nous like en retour (match)
+    const reciprocalLike = await Like.findOne({
+      where: { cat_id1: likedCat, cat_id2: connectedCat },
+    });
+
+    // Créez une nouvelle instance de Like
+    const like = Like.create({
+      cat_id1: connectedCat,
+      cat_id2: likedCat,
+      isMatch: reciprocalLike ? true : null, // Si like reciproque match true, si pas réciproque le laisser à null
```
Comment (jujuck) [inline server/src/cats/cat.resolvers.ts]: ajout de la propriété: isLike : true

File: server/src/cats/cat.resolvers.ts [hunk 6]
```diff
@@ -66,4 +66,95 @@ export default class CatResolver {
...
+    if (!connectedCat || !dislikedCat) {
+      console.error("Mutation like: L'un des chats n'existe pas");
+      return null;
+    }
+
+    // Créez une nouvelle instance de Like
+    const like = Like.create({
+      cat_id1: connectedCat,
+      cat_id2: dislikedCat,
+      isMatch: false,
```
Comment (jujuck) [inline server/src/cats/cat.resolvers.ts:166]: isLike: false

**Review by jujuck — CHANGES_REQUESTED** [review]

**Review by Pas-nou — DISMISSED** [review]

**Review by pjroggy — DISMISSED** [review]

**Review by Acetoshi — COMMENTED** [review]
gg

File: client/src/views/Likes/Likes.tsx [hunk 7]
```diff
@@ -10,7 +10,7 @@ import "./Likes.css";
 
 export const Likes = () => {
   // TODO  ---- En attente de la feature de connexion -----
-  const catId = 5;
+  const catId = 25;
```
Comment (Pas-nou) [inline client/src/views/Likes/Likes.tsx:13]: L'utilisation du contexte sera utilisé sur la PR d'après

**Review by Pas-nou — APPROVED** [review]
Purrfect

**Review by pjroggy — APPROVED** [review]

**Review by jujuck — APPROVED** [review]

**Pull request merged**